apiserver/facades/client/bundle/bundle.go

```diff
@@ -91,3 +99,78 @@ func (b *bundleAPI) GetChanges(args params.BundleChangesParams) (params.BundleCh
 	}
 	return results, nil
 }
+
+// ExportBundle exports the model configuration specified in the argument as bundle.
+func (b bundleAPI) ExportBundle(args params.ExportBundleParams) (params.BundleData, error) {
+	result := params.BundleData{}
+
+	tag, err := names.ParseModelTag(args.ModelTag)
+	if err != nil {
+		return result, errors.Annotate(err, "model-tag cannot be parsed.")
+	}
+
+	st, releaseSt, err := b.state.GetBackend(tag.Id())
+	if err != nil {
+		return result, errors.Trace(err)
+	}
+	defer releaseSt()
+
+	var model description.Model
+	model, err = st.Export()
+	if err != nil {
+		return result, errors.Trace(err)
+	}
+
+	// Fill in the BundleApplicationSpec of BundleData.
+	app := make(map[string]*params.BundleApplicationSpec)
```

Comment:
We probably don't need all these new structs. The end result will just be bundle yaml. That can all be generated here server side by populating the charm.BundleData struct using the model data from Export(). The result of the api call will be a stringresult with the bundleyaml as the string value